recipe/patch_yaml/numpy.yaml

```diff
@@ -0,0 +1,49 @@
+if:
+  has_depends:
+    - numpy?( *)
+    - python >=3.9*
+  timestamp_lt: 1713569710960
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  has_depends:
+    - numpy?( *)
+    - python >=3.10*
+  timestamp_lt: 1713569710960
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  has_depends:
+    - numpy?( *)
+    - python >=3.11*
+  timestamp_lt: 1713569710960
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  has_depends:
+    - numpy?( *)
+    - python >=3.12*
+  timestamp_lt: 1713569710960
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+
+---
+if:
+  has_depends: numpy?( *)
+  subdir_in: noarch
+  timestamp_lt: 1713569710960
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
```

Comment:
This patch doesn't make any sense to me. A noarch python package that just calls `np.sum` on some arrays doesn't need this constraint.

Reply:
That's fair, though I intentionally included noarch packages here, because it's easier IMO to draw a clear boundary between 1.x & 2.x, rather than hope that these packages are not affected in any way, shape or form by the many changes in 2.0 (which aren't just restricted to ABI-changes).

Reply:
The same comment applies to any non-noarch package too.

Reply:
We cannot make the 1.x vs 2.x decision for maintainers. This is too intrusive. They can add their own patches later if they feel they need it.

Reply:
> We cannot make the 1.x vs 2.x decision for maintainers.

I disagree. Perhaps @rgommers can weigh in here.